x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/blockhash/BlockHash.java

```diff
@@ -47,6 +47,9 @@ public abstract sealed class BlockHash implements Releasable, SeenGroupIds //
     /**
      * Add all values for the "group by" columns in the page to the hash and
      * pass the ordinals to the provided {@link GroupingAggregatorFunction.AddInput}.
+     * <p>
+     *     This call will not {@link GroupingAggregatorFunction.AddInput#close} {@code addInput}.
```

Comment:
Were we closing them twice before? I expected another change somewhere after removing the `addInput.close()`

Reply:
Yes indeed! Sorry. I should have called this one out. We have a choice here - either class in the BlockHash or close outside. We were closing twice. Sometimes - it was inconsistent. Some BlockHashs were closing and other weren't.